### ISSUE TITLE
[BUG]: AttributeError: 'TransformerWrapper' object has no attribute '_memory_transform'

### ISSUE THREAD

**Issue by nichoqu** [pycaret/pycaret#2840]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

I used PyCaret 3.0-rc. I developed model and saved them. after almost 3 weeks, now I loaded model. But when I want to use them :

train_predictions = predict_model(model, data=train)

I got the following error:
 AttributeError: 'TransformerWrapper' object has no attribute '_memory_transform'

### Reproducible Example

```python
from pycaret.classification import *
lrm = load_model('/content/drive/MyDrive/Colab/SlopeStability/models/tuned_lrm')
train_predictions = predict_model(lrm, data=pytrain)
```

### Expected Behavior

I should receive   predicted labels.

### Actual Results

```python-traceback
AttributeError                            Traceback (most recent call last)
<ipython-input-18-0eb8688f43df> in <module>()
      1 from pycaret.classification import *
      2  lrm = load_model('/content/drive/MyDrive/Colab/SlopeStability/models/tuned_lrm')
----> 3 train_predictions = predict_model(lrm, data=pytrain)

4 frames
/usr/local/lib/python3.7/dist-packages/pycaret/internal/pipeline.py in __getattr__(self, name)
     86     def __getattr__(self, name: str):
     87         # override getattr to allow grabbing of final estimator attrs
---> 88         return getattr(self._final_estimator, name)
     89 
     90     @property

AttributeError: 'TransformerWrapper' object has no attribute '_memory_transform'
```

### Installed Versions

<details>
Replace this line with the output of the version code above.
</details>

**Comment by tvdboom** [comment]
I can't debug the error without a full working example. Please, either share the model file or the code to produce that file.

**Comment by ngupta23** [comment]
It is likely that you trained the model with a different version of pycaret than the one you are using for prediction now.

**Comment by jnsflt** [comment]
I've noticed a similar error as well when scikit-learn released a new version a couple of days ago. When I trained the model, I used scikit-learn==1.1.1, but after creating a docker-image for my project (re-installing every dependency) I used scikit-learn==1.1.2, thus causing an error. 

I know its of course up to the user to know that their versions is alike, but since PyCaret doesn't specify an exact library-version in its `requirements.txt` file, it will probably cause some errors again.

Is there any reason why you dont keep exact version in your requirements?

**Comment by jnsflt** [comment]
Funny coincidence, but `plotly-resampler` released a new version (0.7.2.3) one hour ago and now I cant re-install PyCaret, since the `requirements.txt` says `plotly-resampler>=0.7.2.2` and there is probably a bug from `plotly-resampler` which makes it not able to install.

**Comment by ngupta23** [comment]
> Is there any reason why you don't keep exact version in your requirements?

It is never a good practice to hard code the version in open source since users will be using different systems, installers, version of python, etc. and the availability of the library versions will vary based on that. In order to make sure that the open source library works for a wide spectrum of users, usually library versions are not pinned and it is up to the installer (pip, conda, etc.) to resolve the right version. 

In addition, once the installer has installed a version, it is usually a good practice on the user's end to pin the installation version on their end to the one that got installed to avoid the issues that you mentioned above.

For example, if pycaret 3.0.0 mentions sklearn > 1.0.0 and you install pycaret. Your installer may install sklearn 1.1.0 since it satisfies the condition >1.0.0. In this case, the user can create a requirements.txt file on their end pinning sklearn to 1.1.0 and pycaret to 3.0.0 to create a reproducible install in the future.

**Comment by ngupta23** [comment]
> Funny coincidence, but `plotly-resampler` released a new version (0.7.2.3) one hour ago and now I cant re-install PyCaret, since the `requirements.txt` says `plotly-resampler>=0.7.2.2` and there is probably a bug from `plotly-resampler` which makes it not able to install.

I don't see a new release in the last day. Where is this new release?

https://pypi.org/project/plotly-resampler/#history

**Comment by jnsflt** [comment]
> > Funny coincidence, but `plotly-resampler` released a new version (0.7.2.3) one hour ago and now I cant re-install PyCaret, since the `requirements.txt` says `plotly-resampler>=0.7.2.2` and there is probably a bug from `plotly-resampler` which makes it not able to install.
> 
> I don't see a new release in the last day. Where is this new release?
> 
> https://pypi.org/project/plotly-resampler/#history

Thank you for the clarification above! 

Maybe they removed the release (if thats even possible). I saw it at https://github.com/predict-idlab/plotly-resampler when it was "1 hour ago". But as I said, something was wrong so maybe they removed it.
